src/pages/Account/FirebaseEmailHandler/PasswordReset.tsx

```diff
@@ -0,0 +1,177 @@
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  // @see
+  // https://firebase.google.com/docs/auth/custom-email-handler
+  handleResetPassword = (actionCode: string) => {
+    // Verify the password reset code
+    // is valid.
+    auth
+      .verifyPasswordResetCode(actionCode)
+      .then(() => {
+        this.setState({ showPasswordResetForm: true });
+      })
+      .catch(() => {
+        this.setState({
+          hasError: true,
+          errorMessage: 'Invalid or expired action code. Please reset the password again.',
+        });
+      });
+  }
+
+  handlePasswordResetSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!this.props.oobCode) {
+      return this.setState({
+        hasError: true,
+        errorMessage: 'Invalid or expired action code. Please reset the password again.',
+      });
+    }
+
+    if (!this.state.password) {
+      return alert('Invalid password.');
+    }
+
+    // Save the new password.
+    auth
+      .confirmPasswordReset(this.props.oobCode, this.state.password)
+      .then(() => {
+        this.setState({ showPasswordResetForm: false, showPasswordResetSuccess: true });
+      }).catch((err: Error) => {
+        console.error(err);
+        this.setState({
+          hasError: true,
+          errorMessage: 'Password is too weak(at least 8 characters) or the password reset link has expired.',
+        });
+      });
+  }
+
+  renderError(): React.ReactNode {
+    return (
+      <Container tag={Fade} className={FIREBASE_EMAIL_HANDLER_CONTAINER_CLASSES}>
+        <h2 className={FIREBASE_EMAIL_HANDLER_TITLE_CLASSES}>Sorry, there was an error</h2>
+        <p className={FIREBASE_EMAIL_HANDLER_SUBTITLE_CLASSES}>{this.state.errorMessage}</p>
+        <Row>
+          <a target="_blank" href="https://support.beenest.com/">
+            <Button color="primary" className={FIREBASE_EMAIL_HANDLER_BUTTON}>Contact us for further help.</Button>
+          </a>
+        </Row>
+      </Container>
+    );
+  }
+
+  renderPasswordResetForm(): React.ReactNode {
+    return (
+      <Container tag={Fade} className={FIREBASE_EMAIL_HANDLER_CONTAINER_CLASSES}>
+        <Col md="6" lg="5">
+          <h2 className={FIREBASE_EMAIL_HANDLER_TITLE_CLASSES}>Reset Password</h2>
+          <Form className="mt-4" onSubmit={this.handlePasswordResetSubmit} type="POST">
+            <FormGroup>
+              <Label for="password" className="form-label">
+                Password
+              </Label>
+              <Input
+                type="password"
+                name="password"
+                value={this.state.password}
+                id="password"
+                onChange={this.handleFormChange}
+                placeholder="********"
+              />
+            </FormGroup>
+            <Button color="primary" className={`${FIREBASE_EMAIL_HANDLER_BUTTON} w-100 mt-3`} type="submit">Save</Button>
```

Comment:
likewise here, can this just be a generic button instead a custom one for firebase?